core/gdxsv/gdxsv.cpp

```diff
@@ -146,6 +146,7 @@ void Gdxsv::Update() {
         // Don't edit memory at vsync if ggpo::active
         WritePatch();
     }
+    WriteWidescreenPatchDisk2();
```

Comment:
Doesn't GGPO get confused if rewrite memory at this timing?

Reply:
The string "GGPO Active" is being logged in `Gdxsv::Update()`

<img width="988" alt="image" src="https://user-images.githubusercontent.com/602245/193492633-358f6d89-5dc1-4e18-bfc9-154e244c7fdd.png">
There isn't any `GDX_RPC_SOCK_WRITE`(`GdxsvBackendRollback::OnSockWrite`) at the very end of the briefing screen

(95 vblanks to be precise)

There are 4 vblanks before the first `GdxsvBackendRollback::OnSockWrite` during battle start
![image](https://user-images.githubusercontent.com/602245/193492979-fc97cdcc-25f9-4f7b-b404-5711f7895140.png)
I have increased the contrast for the first frame, this explains why the widescreen patch failed, since it is patched later than the actual draw

If there isn't any traffic, are we safe since no rollback should happen?


Relevant log:
```
gdxsv/gdxsv_backend_rollback.cpp:546 N[COMMON]: GdxsvBackendRollback::ApplyPatch
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv_backend_rollback.cpp:546 N[COMMON]: GdxsvBackendRollback::ApplyPatch
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv_backend_rollback.cpp:546 N[COMMON]: GdxsvBackendRollback::ApplyPatch
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv_backend_rollback.cpp:546 N[COMMON]: GdxsvBackendRollback::ApplyPatch

//
// The very end of briefing screen
//
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active

//
// Right before the first frame is drawn
//
gdxsv/gdxsv.cpp:693 E[COMMON]: Write Widescreen Patch

//
// 4 more vblanks, 1st frame is drawn here
//
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active

//
// First ApplyPatch in battle, too late for widescreen patch
//
gdxsv/gdxsv_backend_rollback.cpp:546 N[COMMON]: GdxsvBackendRollback::ApplyPatch
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
gdxsv/gdxsv_backend_rollback.cpp:546 N[COMMON]: GdxsvBackendRollback::ApplyPatch
gdxsv/gdxsv.cpp:150 N[COMMON]: GGPO Active
...
```

Reply:
I'll see how it goes, but we should have a better widescreen patch. The current state is too dirty.